lib/matplotlib/axes/_base.py

```diff
@@ -2430,6 +2430,16 @@ def handle_single_axis(scale, autoscaleon, shared_axes, interval,
                 dl.extend(y_finite)
 
             bb = mtransforms.BboxBase.union(dl)
+            # fall back on the viewlimits if this is not finite:
+            vl = None
+            if not np.isfinite(bb.intervalx).all():
+                vl = mtransforms.BboxBase.union([ax.viewLim for ax in shared])
+                bb.intervalx = vl.intervalx
+            if not np.isfinite(bb.intervaly).all():
+                if vl is None:
```

Comment:
I would have just stupidly recalculated `vl`. :smile: This only happens for the degenerate case that `bb.intevalx` and `bb.intervaly` contain non-finite values. IMHO preferable due to symmetry and simpler logic of the code.